pinot-common/src/main/java/org/apache/pinot/common/utils/LLCSegmentName.java

```diff
@@ -127,6 +181,11 @@ public String getSegmentName() {
   public int compareTo(LLCSegmentName other) {
     Preconditions.checkArgument(_tableName.equals(other._tableName),
         "Cannot compare segment names from different table: %s, %s", _segmentName, other.getSegmentName());
+    String thisTopicName = _topicName == null ? "" : _topicName;
+    String otherTopicName = other._topicName == null ? "" : other._topicName;
+    if (!thisTopicName.equals(otherTopicName)) {
+      return StringUtils.compare(_topicName, other._topicName);
+    }
```

Comment:
Do you want to compare topic name here? What is the side effect of comparing it?